File: solutions/typescript/126-meeting-rooms.ts
/**
 * Problem 126: Meeting Rooms (LeetCode 252)
 * Difficulty: Easy
 * Language: TypeScript
 */
function canAttendMeetings(intervals: number[][]): boolean {
  const sorted = [...intervals].sort((a, b) => a[0] - b[0]);
  return sorted.every((v, i) => i === 0 || sorted[i - 1][1] <= v[0]);
}

File: solutions/clojure/126-meeting-rooms.clj
;; Problem 126: Meeting Rooms (LeetCode 252)
;; Difficulty: Easy
;; Language: Clojure
;; 
(defn can-attend-meetings [intervals]
  (let [sorted (sort-by first intervals)]
    (every? (fn [[[_ e] [s _]]] (<= e s))
            (partition 2 1 sorted))))
